### ISSUE TITLE
Testing also for http when https is specified breaks OONI mobile

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#1685]
As reported by @lorenzoprimi, this breaks OONI mobile in two ways:

1. because it return URLs that are not in the pre-populated database (i.e. `http://example.com` is not in the database while `https://example.com` is)

2. because it canonicalises the URLs, therefore `https://example.com` is written in the database but MK returns the canonical `https://example.com/`

**Comment by bassosimone** [comment]
I'll try to sum up the discussion we've had yesterday with @hellais:

1. URLs should be canonicalised in [citizenlab/test-lists](https://github.com/citizenlab/test-lists) and in [ooni/orchestra](https://github.com/ooni/orchestra): _be strict in what you send_

2. MK should not emit the canonicalised URL to the app, but rather the original URL as long as it parses fine (and URL that doesn't parse fine are anyway deferred to web connectivity): _be liberal in what you accept_

3. the two above points should prevent issues with https URLs passed to MK by the app being then not recognised by the app because MK has canonicalised them

4. the app should, for now, ignore tests for http URLs that are not expected, we will do something about it later (@hellais can you expand on this point? I don't remember any specific action item coming out of this part of the discussion)

**Comment by bassosimone** [comment]
I've committed the fix mentioned above for both master and stable. When the relevant PRs will be created and merged, we'll have fixed the MK part of the issue. Still, it's a bit too early to close the issue, because there are more bits of the issue itself that need to be addressed.

**Comment by bassosimone** [comment]
@lorenzoPrimi right. On your end, you probably need to change the code with @xanscale such that events for unrecognised input are ignored. What's the proper repo to put such issue in?

**Comment by bassosimone** [comment]
I meant, when you receive an event for an URL that you did not submit to MK

**Comment by lorenzoPrimi** [comment]
What should I do in this case? add the url to the db with category MISC and cc XX or ignore it? @hellais

**Comment by bassosimone** [comment]
Ignore in v2.0. We need to think about a better strategy for v2.1.

**Comment by bassosimone** [comment]
@lorenzoPrimi this issue has been completed on all sides except on the mobile app. Let me know when the mobile app has learned to ignore unknown URLs and we're good.

**Comment by lorenzoPrimi** [comment]
Waiting for the new pod to check it

**Comment by bassosimone** [comment]
As mentioned above, we _should_ discuss this for v0.10.0.

**Comment by lorenzoPrimi** [comment]
I’m ok with whatever you want to do, it’s more a philosophical discussion rather than a techincal one. You have to decide if it’s more important to test two different websites vs one website in its two version (http&https)

**Comment by lorenzoPrimi** [comment]
I also think this can be achieved orchestra `/urls` side by sending the http and https version of the website when needed

**Comment by lorenzoPrimi** [comment]
Also maybe we want to revisit the UX if testing the same website `http` and `https` and not having them randomly in the list @elioqoshi

**Comment by hellais** [comment]
I think the matter also is that if we do test both the HTTP and the HTTPS version of the site, the app should in some way “aggregate” these two results into a single view
And it’s a bit tricky because how it was working before, they would end up being both inside of a separate measurement row so the aggregation needs to happen at the app level.

This may also be related to: https://github.com/ooni/spec/issues/92 & https://github.com/measurement-kit/measurement-kit/issues/1422

**Comment by bassosimone** [comment]
Postponing to v0.11.0 per the discussion with @hellais on Slack

**Comment by bassosimone** [comment]
@elioqoshi:
> Is there a decision made on how to handle this from a practical view? Would the app do the aggregation like @hellais suggested for http and https versions?

I think it would be fine if the engine did not aggregate and just tested what is told to test, and the app then could do or not do aggregation. I believe it's smarter for the API to provide us the right URLs depending on the measurement policy we want to enforce, and to possibly do the same at the OONI Run level, where we automatically expand `https://` URLs to also test `http://` URLs. At this point the only use case which is not covered is the one of a manually inserted URL. Also in this case, the app may just basically expand the URLs and tell the user: "look, we're testing both, because this gives us more information". This should cover all the possible cases.

Honestly, I'd be fine if we just do that in the API and in OONI Run. I would also recommend proceeding piecemeal and fixing all these issues in separate steps, so we move more closely towards what we want, rather than having this issue open and rot until we find out the perfect solution (TM).

That said, we can discuss further and I do not reckon this is a very important hill for me to die.

**Comment by elioqoshi** [comment]
Understood Feel free to loop me in once there is a spec decided to move forward with this.
